configure.ac

```
@@ -444,10 +444,20 @@ if test "x$with_dl" != "xno"; then
 fi
 
 ## check for atomics library before anything that might need it
-# AC_SEARCH_LIBS pollutes LIBS
 SQUID_STATE_SAVE(LIBATOMIC)
-AC_SEARCH_LIBS([__atomic_load_8],[atomic],[
-  test "$ac_res" = "none required" || ATOMICLIB=$ac_res],[])
+AC_MSG_CHECKING(library containing __atomic_is_lock_free)
+LIBS="-latomic"
```

This is not my area of expertise, but I would expect us to _append_ to LIBS (just in case) instead of resetting it. I know we do something like that for eCAP, OpenSSL, and GnuTLS already.

```suggestion
LIBS="$LIBS -latomic"
```